### PR TITLE
COM-583 Add company contract start date check on contract creation

### PR DESCRIPTION
**Pull request by jctrebalag**

### CODE REVIEW

File: helpers/contracts.js [hunk 1]
```diff
@@ -16,6 +16,37 @@ const { CUSTOMER_CONTRACT, COMPANY_CONTRACT } = require('./constants');
 const { createAndReadFile } = require('./file');
 const ESignHelper = require('../helpers/eSign');
 const EventRepository = require('../repositories/EventRepository');
+const ContractRepository = require('../repositories/ContractRepository');
+
+
+exports.createContract = async (contractPayload) => {
+  if (contractPayload.status === COMPANY_CONTRACT) {
+    const endingCompanyContracts = await ContractRepository.getUserEndingCompanyContracts(contractPayload.user, contractPayload.startDate);
```
Comment (sophiemoustard) [inline helpers/contracts.js]: end**ed**

Comment (sophiemoustard) [reply helpers/contracts.js]: pour endingCompanyContracts et getUserEndingCompanyContracts

File: helpers/contracts.js [hunk 2]
```diff
@@ -16,6 +16,37 @@ const { CUSTOMER_CONTRACT, COMPANY_CONTRACT } = require('./constants');
...
+    if (endingCompanyContracts.length && moment(contractPayload.startDate).isBefore(moment(endingCompanyContracts[0].endDate), 'd')) {
```
Comment (sophiemoustard) [inline helpers/contracts.js]: isSameOrBefore

File: helpers/contracts.js (hunk 1 above)
Comment (jctrebalag) [reply helpers/contracts.js]: Done !

File: helpers/contracts.js (hunk 2 above)
Comment (jctrebalag) [reply helpers/contracts.js]: Done !

**Pull request merged**